### PR TITLE
add support for microsoft wsl for docker driver

### PR DESCRIPTION
**Pull request by medyagh**
have not tested myself but acording to this comment it works 
https://github.com/kubernetes/minikube/issues/5392#issuecomment-638562724

closes https://github.com/kubernetes/minikube/issues/7420 
closes https://github.com/kubernetes/minikube/issues/5392

### CODE REVIEW

File: pkg/minikube/driver/driver.go [hunk 1]
```diff
@@ -131,7 +131,14 @@ func NeedsRoot(name string) bool {
...
 func NeedsPortForward(name string) bool {
 	// Docker for Desktop
-	return IsKIC(name) && (runtime.GOOS == "darwin" || runtime.GOOS == "windows")
+	return IsKIC(name) && (runtime.GOOS == "darwin" || runtime.GOOS == "windows" || isMicrosoftWSL())
+}
+
+// isMicrosoftWSL will return true if process is running in WSL in windows
+// checking for WSL env var based on this https://github.com/microsoft/WSL/issues/423#issuecomment-608237689
+// also based on https://github.com/microsoft/vscode/blob/90a39ba0d49d75e9a4d7e62a6121ad946ecebc58/resources/win32/bin/code.sh#L24
+func isMicrosoftWSL() bool {
```
Comment (priyawadhwa) [inline pkg/minikube/driver/driver.go:140]: Let's make this function public so that we don't need to repeat it in the integration package

Comment (medyagh) [reply pkg/minikube/driver/driver.go:140]: sounds good

**Pull request merged**